cfme/utils/ssh.py

```diff
@@ -309,18 +306,34 @@ def run_command(
             stderr = session.makefile_stderr()
             while True:
                 if session.recv_ready:
-                    for line in stdout:
-                        output.append(line)
-                        if self._streaming:
-                            self.f_stdout.write(line)
+                    line = stdout.next()
+                    output.append(line)
+                    if self._streaming:
+                        self.f_stdout.write(line)
 
                 if session.recv_stderr_ready:
-                    for line in stderr:
-                        output.append(line)
-                        if self._streaming:
-                            self.f_stderr.write(line)
+                    line = stderr.next()
+                    output.append(line)
+                    if self._streaming:
+                        self.f_stderr.write(line)
 
                 if session.exit_status_ready():
+                    while session.recv_ready:
```

Comment:
A comment on why this is needed will help future folks who look at this code (could even be future me and you).